src/Api/QueryParameterValidator/QueryParameterValidator.php

```diff
@@ -60,8 +61,14 @@ public function validateFilters(string $resourceClass, array $resourceFilters, a
             }
 
             foreach ($filter->getDescription($resourceClass) as $name => $data) {
-                foreach ($this->validators as $validator) {
-                    if ($errors = $validator->validate($name, $data, $queryParameters)) {
+                $collectionFormat = $this->getCollectionFormat($data);
+
+                // validate simple values
+                $errorList[] = $this->validate($name, $data, $queryParameters);
+
+                // manipulate query data to validate each value
+                foreach ($this->iterateValue($name, $queryParameters, $collectionFormat) as $scalarQueryParameters) {
```

Comment:
We iterate over each splitted value and run all validators against the singular value